mlflow/transformers.py

```diff
@@ -0,0 +1,799 @@
+        conf[_PROCESSOR_KEY] = _load_component(
+            local_path, _PROCESSOR_KEY, flavor_config[_PROCESSOR_TYPE_KEY]
+        )
+
+    for component_key in flavor_config[_COMPONENTS_BINARY_KEY]:
+        component_type_key = f"{component_key}_type"
+        component_type = flavor_config[component_type_key]
+        conf[component_key] = _load_component(local_path, component_key, component_type)
+    return conf
+
+
+def _hub_access():
+    """
+    Wrapper around importing huggingface_hub for ModelCard retrieval, providing access to the
+    module if the library is installed, else noop.
+    """
+    import importlib
+
+    try:
+        return importlib.import_module("huggingface_hub")
+    except ImportError:
+        pass
+
+
+def _fetch_model_card(model_or_pipeline):
+    """
+    Attempts to retrieve the model card for the specified model architecture iff the
+    `huggingface_hub` library is installed. If a card cannot be found in the registry or
+    the library is not installed, returns None.
+    """
+    from transformers import Pipeline
+
+    model = (
+        model_or_pipeline.model if isinstance(model_or_pipeline, Pipeline) else model_or_pipeline
+    )
+    card_loader = _hub_access()
+    if card_loader:
+        return card_loader.ModelCard.load(model.name_or_path)
+
+
+def _build_pipeline_from_model_input(model, task: str):
+    """
+    Utility for generating a pipeline from component parts.
+    """
+    from transformers import PreTrainedModel, TFPreTrainedModel, FlaxPreTrainedModel, pipeline
+
+    if not isinstance(model, (PreTrainedModel, TFPreTrainedModel, FlaxPreTrainedModel)):
+        raise MlflowException(
+            f"The provided model is not the correct type. The type provided is: {type(model)}. "
+            f"The model must inherit from one of: PreTrainedModel, TFPreTrainedModel, "
+            f"FlaxPreTrainedModel"
+        )
+
+    model_architecture_name = model.name_or_path
+
+    pipeline_config = {
+        "task": task,
+        "model": model,
+        **_configure_extractors(model_architecture_name),
+    }
+
+    return pipeline(**pipeline_config)
+
+
+def _configure_extractors(architecture):
+    """
+    Performs a hierarchy-based acquisition of an appropriate processor or
+    feature extractor based on a named and registered model type within the HuggingFace repo.
+    In order to support as many use cases as possible, all available processors and extractors
+    are loaded.
+    """
+    import transformers
+
+    extractor_types = {
+        "tokenizer": "AutoTokenizer",
+        "image_processor": "AutoImageProcessor",
+        "feature_extractor": "AutoFeatureExtractor",
+    }
+    extractors = {}
+    for extractor_type, loader in extractor_types.items():
+        try:
+            instance = getattr(transformers, loader)
+            extractors[extractor_type] = instance.from_pretrained(architecture)
+        except (KeyError, OSError):
```

Comment:
Curious what causes `KeyError`?

Reply:
The OSError is thrown if the object has a cache reference (pre-loaded) after a call to `.from_pretrained()` while the KeyError happens if there isn't a local copy and the `.from_pretrained()` fetches from their repository. 

I've removed this code in my next commit since we're moving to explicitly defined component definitions due to correctness issues with loading pretrained component objects (tokenizers, image_processors, etc.) on a retrained model. We're no longer fetching base extractors from the hub; users either have to explicitly provide them or use a Pipeline.